platform/darwin/src/MGLMapSnapshotter.mm

```diff
@@ -33,23 +35,61 @@
 const CGFloat MGLSnapshotterMinimumPixelSize = 64;
 
 
-@interface MGLMapSnapshotOverlay()
-
-- (instancetype)initWithContext:(CGContextRef)context;
+@interface MGLMapSnapshotOverlay() <MGLMapSnapshotProtocol>
+@property (nonatomic, assign) CGFloat scale;
+- (instancetype)initWithContext:(CGContextRef)context scale:(CGFloat)scale pointForFn:(mbgl::MapSnapshotter::PointForFn)pointForFn latLngForFn:(mbgl::MapSnapshotter::LatLngForFn)latLngForFn;
 
 @end
 
-@implementation MGLMapSnapshotOverlay
+@implementation MGLMapSnapshotOverlay {
+    mbgl::MapSnapshotter::PointForFn _pointForFn;
+    mbgl::MapSnapshotter::LatLngForFn _latLngForFn;
+}
 
-- (instancetype) initWithContext:(CGContextRef)context {
+- (instancetype) initWithContext:(CGContextRef)context scale:(CGFloat)scale pointForFn:(mbgl::MapSnapshotter::PointForFn)pointForFn latLngForFn:(mbgl::MapSnapshotter::LatLngForFn)latLngForFn {
     self = [super init];
     if (self) {
         _context = context;
+        _scale = scale;
+        _pointForFn = pointForFn;
+        _latLngForFn = latLngForFn;
     }
 
     return self;
 }
 
+#if TARGET_OS_IPHONE
+
+- (CGPoint)pointForCoordinate:(CLLocationCoordinate2D)coordinate
+{
+    mbgl::ScreenCoordinate sc = _pointForFn(MGLLatLngFromLocationCoordinate2D(coordinate));
+    return CGPointMake(sc.x, sc.y);
+}
+
+- (CLLocationCoordinate2D)coordinateForPoint:(CGPoint)point
+{
+    mbgl::LatLng latLng = _latLngForFn(mbgl::ScreenCoordinate(point.x, point.y));
+    return MGLLocationCoordinate2DFromLatLng(latLng);
+}
+
+#else
+
+- (NSPoint)pointForCoordinate:(CLLocationCoordinate2D)coordinate
+{
+    mbgl::ScreenCoordinate sc = _pointForFn(MGLLatLngFromLocationCoordinate2D(coordinate));
+    CGFloat height = ((CGFloat)CGBitmapContextGetHeight(self.context))/self.scale;
+    return NSMakePoint(sc.x, height - sc.y);
+}
+
+- (CLLocationCoordinate2D)coordinateForPoint:(NSPoint)point
+{
+    CGFloat height = ((CGFloat)CGBitmapContextGetHeight(self.context))/self.scale;
+    auto screenCoord = mbgl::ScreenCoordinate(point.x, height - point.y);
```

Comment:
Thanks for accounting for the difference in coordinate system origin on macOS! ❤️